android/modules/media/src/java/ti/modules/titanium/media/MediaModule.java

```diff
@@ -1123,14 +1125,14 @@ protected static KrollDict createDictForImage(String path)
 				parcelFileDescriptor.close();
 				imageData = TiBlob.blobFromImage(image);
 			} catch (FileNotFoundException e) {
-				imageData = createImageData(parts, null);
+				imageData = createImageData(parts, MIME_IMAGE);
 			} catch (IOException e) {
-				imageData = createImageData(parts, null);
+				imageData = createImageData(parts, MIME_IMAGE);
 			}
 		} else {
-			imageData = createImageData(parts, null);
+			imageData = createImageData(parts, MIME_IMAGE);
 		}
-		return createDictForImage(imageData, null);
+		return createDictForImage(imageData, MIME_IMAGE);
```

Comment:
We shouldn't hard code the mime-type like this. This is because the end-user is capable of selecting files others than images from the gallery. I've done it on my devices. In this case, a zero width/height makes sense.

Have a look at how my open PR handles it here...
[https://github.com/appcelerator/titanium_mobile/pull/9223/files-diff](https://github.com/appcelerator/titanium_mobile/pull/9223/files#diff-32e2c2f2c365eee27a85d7bd7c08e1fcR1120)

Here is a code snipped from my PR. It'll fetch the mime-type from the gallery app's content provider using the given content URI. I've tested this with photos, videos, and text files and it works. And I think it's okay to leave it `null` if the selected file is unknown since that won't happen for photos.

```java
// Determine the mime type for the given file.
String mimeType = null;
boolean isPhoto = false;
try {
	mimeType = TiApplication.getInstance().getContentResolver().getType(Uri.parse(path));
	if ((mimeType != null) && mimeType.toLowerCase().startsWith("image")) {
		isPhoto = true;
	}
} catch (Exception ex) {
}
```

Reply:
Also note that the above will correctly set the mime-type for the selected image format such as "image/png", "image/jpeg", etc.